core/src/main/scala/com/softwaremill/react/kafka_api_producer/Producer.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_producer
+
+import scala.concurrent.Future
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Sink
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.clients.producer.RecordMetadata
+import com.softwaremill.react.kafka_api_opt3.Consumer
+
+object Producer {
+
+  def flow[K, V](producerProvider: () => KafkaProducer[K, V]): Flow[ProducerRecord[K, V], Future[ProducerRecord[K, V]], NotUsed] =
```

Comment:
@RayRoestenburg ah, that was the missing piece, thanks. It should of course be part of the output.

Errors, isn't the stream simply completed with failure?